src/input-parameters.ts

```diff
@@ -1,61 +1,53 @@
-import { getBooleanInput, getMultilineInput, getInput } from '@actions/core'
+import { getMultilineInput, getInput } from '@actions/core'
+import { OverwriteMode } from '@octopusdeploy/api-client'
+
+const EnvironmentVariables = {
+  URL: 'OCTOPUS_URL',
+  ApiKey: 'OCTOPUS_API_KEY',
+  Space: 'OCTOPUS_SPACE'
+} as const
 
 export interface InputParameters {
-  // required parameters
-  packages: string[]
-  apiKey: string
   server: string
+  apiKey: string
   space: string
-  // optional parameters
-  debug?: boolean
-  logLevel: string
-  overwriteMode: string
-  proxy: string
-  proxyPassword: string
-  proxyUsername: string
-  timeout: string
-  useDeltaCompression: boolean
+  packages: string[]
+  overwriteMode: OverwriteMode
 }
 
-export function getInputParameters(): InputParameters {
-  return {
-    // required parameters
-    packages: getMultilineInput('packages'),
-    apiKey: getInput('api_key'),
-    server: getInput('server'),
-    space: getInput('space'),
-    // optional parameters
-    debug: getBooleanInput('debug'),
-    logLevel: getInput('log_level'),
-    overwriteMode: getInput('overwrite_mode'),
-    proxy: getInput('proxy'),
-    proxyPassword: getInput('proxy_password'),
-    proxyUsername: getInput('proxy_username'),
-    timeout: getInput('timeout'),
-    useDeltaCompression: getBooleanInput('use_delta_compression')
+export function getInputParameters(isRetry: boolean): InputParameters {
+  const overwriteMode: OverwriteMode =
+    (OverwriteMode as any)[getInput('overwrite_mode')] || // eslint-disable-line @typescript-eslint/no-explicit-any
```

Comment:
I would normally see something like this:
```suggestion
    OverwriteMode[getInput('overwrite_mode') as keyof typeof OverwriteMode] ||
```
You could even move it to a type `type OverwriteModeKey = keyof typeof OverwriteMode` and use it like so `OverwriteMode[getInput('overwrite_mode') as OverwriteModeKey]`